### PR TITLE
Removed references to MongoDB and updated emitter to transmit a rest …

### PR DESCRIPTION
**Pull request by moffatted**
…call to an external host to launch a terraformed jenkins server.  Added logging, and properties configuration.  Also updated Dockerfile to accept JAVA_OPTS and run docker-entrypoint script.  Probably still more todo, but wanted to get checked in what I have so far.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -18,51 +18,100 @@
...
+     <parent>
         <groupId>cms.sre</groupId>
         <artifactId>spring-parent-pom</artifactId>
-        <version>1.0</version>
-    </parent>
+        <version>1.1.13</version>
+	 </parent>
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <start-class>cms.sre.gitlab_webhook.App</start-class>
```
Comment (davidhhessler) [inline pom.xml]: This is not needed.  Please remove

File: src/main/java/cms/sre/gitlab_webhook/health/HealthCheck.java [hunk 2]
```diff
@@ -1,29 +1,18 @@
...
         Health health = null;
-        if (this.mongoTemplate != null) {
-            health = new MongoHealthIndicator(this.mongoTemplate)
-                    .health();
-        } else {
-            health = Health.down()
-                    .withDetail("MongoTemplate-Autowired", false)
-                    .build();
-        }
+
```
Comment (davidhhessler) [inline src/main/java/cms/sre/gitlab_webhook/health/HealthCheck.java]: You need to have a health check.  Things to check:
-- Are all config value hydrated correct?

File: src/test/java/cms/sre/gitlab_webhook/TestConfiguration.java [hunk 3]
```diff
@@ -1,53 +1,11 @@
...
-import de.flapdoodle.embed.mongo.config.MongodConfigBuilder;
-import de.flapdoodle.embed.mongo.config.Net;
-import de.flapdoodle.embed.mongo.distribution.Version;
-import de.flapdoodle.embed.process.runtime.Network;
 import org.springframework.core.Ordered;
 import org.springframework.core.annotation.Order;
 
 @org.springframework.boot.test.context.TestConfiguration
 @Order(value = Ordered.HIGHEST_PRECEDENCE)
 public class TestConfiguration extends App {
```
Comment (davidhhessler) [inline src/test/java/cms/sre/gitlab_webhook/TestConfiguration.java]: You can delete this file now.

File: src/test/java/cms/sre/gitlab_webhook/utils/WebHookValidationUtil.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+public class WebHookValidationUtil {
```
Comment (davidhhessler) [inline src/test/java/cms/sre/gitlab_webhook/utils/WebHookValidationUtil.java]: Fix this.

**Comment by moffatted** [comment]
Recommended changes noted.

**Pull request closed without merging**